doc/notes/SPIcoordinates.tex

```diff
@@ -87,16 +87,49 @@
             {\sqrt{\left(\vec{\rho}\cdot\hat{x}\right)^2 + \left(\vec{\rho}\cdot\hat{z}\right)^2}}
         %
         = \frac{y\,\mathrm{sgn}\left(R-R_0\right)}{\sqrt{
-            \left(x-R_0-\frac{(x-R_0)R_0}{\sqrt{\left(x-R_0\right)^2 + z^2}}\right)^2
-            + \left(z-\frac{zR_0}{\sqrt{\left(x-R_0\right)^2 + z^2}}\right)^2
+            \left(x+R_0-\frac{(x+R_0)R_0}{\sqrt{\left(x+R_0\right)^2 + z^2}}\right)^2
+            + \left(z-\frac{zR_0}{\sqrt{\left(x+R_0\right)^2 + z^2}}\right)^2
         }},
     \end{equation}
     with
     \begin{equation}
-        R = \sqrt{(x-R_0)^2+z^2}.
+        R = \sqrt{(x+R_0)^2+z^2}.
     \end{equation}
     The sign function appears since
     $\sqrt{(\vec{\rho}\cdot\hat{x})^2 + (\vec{\rho}\cdot\hat{z})^2}$
     should correspond to the horizontal cartesian coordinate in the poloidal
     plane (which is negative in the second and third quadrants).
+    
+    \section{Cylindrical coordinates to analytic B}
+    The conversion from cartesian coordinates to flux surface coordinates in the analytical B-field geometry in DREAM is most easily done via a transformation to cylindrical coordinates. The reason for this is that the poloidal angle $\theta$ does not have a simple geometrical definition in this context.
+    
+    The flux surface geometry in the analytic B mode is related to the cylindrical coordinates according to
+    \begin{align}
+R &= R_0 + \Delta(r) + r\cos[\theta +\delta(r) \sin\theta], \nonumber \\
+Z &= r \kappa(r) \sin\theta,
+\label{eq:analyticB system}
+\end{align}
+where $\kappa$ is the elongation, $\Delta$ the Shafranov shift and $\delta$ a triangularity parameter. We solve this equation system by eliminating the $\theta$-dependence from the equation for $R$ using the equation for $Z$, and in that way obtain an equation for $r$ alone which is straightforward to solve numerically. First, we rewrite the equation for $R$ in terms of $\cos{\theta}$ and $\sin{\theta}$ rather than $\theta$ itself:
+	\begin{equation}
+		R = R_0 + \Delta (r) + r[\cos{\theta}\cos{(\delta(r)\sin{\theta})} - \sin{\theta}\sin{(\delta(r)\sin{\theta})}].
+		\label{eq:RAnalyticB}
+	\end{equation}
+	From the equation for $Z$, we have
+	\begin{align}
+		\sin{\theta} &= \frac{Z}{r\kappa(r)}\nonumber \\
+		\cos{\theta} &= \pm \sqrt{1-\left(\frac{Z}{r\kappa(r)}\right)^2}.
+		\label{eq:SinCosAnalyticB}
+	\end{align}
+	It remains now to determine the sign on $\cos{\theta}$. We do this by calculating the major radius $R_\mathrm{crit}$ where $\cos{\theta}=0$. As the major radius is a monotonically increasing function of $\cos{\theta}$ {\color{red}(which it always is, right?)}, if $R>R_\mathrm{crit}$, $\cos{\theta}>0$, and vice versa. The critical major radius is calculated by noting that when $\cos{\theta}=0$, we have $\sin{\theta}=1$, so that
```

Comment:
Technically R can vary non-monotonically with cos(theta), but this would most likely violate other assumptions in our bounce-averaged formulation (we assume that B varies monotonically between the two extrema on a flux surface). For fun, I created the attached figure, which has delta=1.5 and a non-monotonic R.
![ExcessiveDelta](https://user-images.githubusercontent.com/10938436/123808136-c5f74780-d8f0-11eb-8cf0-2580857d857d.png)